R/generateSettings.R

```diff
@@ -36,87 +40,78 @@ generateSettings <- function(standard="None", chart="eDish", partial=FALSE, part
-  }
+  standard_low <- tolower(standard)
   
-  if(tolower(standard)=="sdtm"){
-    potential_settings[["id_col"]]<-"USUBJID"
-    potential_settings[["value_col"]]<-"STRESN"
-    potential_settings[["measure_col"]]<-"TEST"
-    potential_settings[["normal_col_low"]]<-"STNRLO"
-    potential_settings[["normal_col_high"]]<-"STNRHI"
-    potential_settings[["studyday_col"]]<-"DY"
-    potential_settings[["visit_col"]]<-"VISIT"
-    potential_settings[["visitn_col"]]<-"VISITNUM"
-    potential_settings[["measure_values"]][["ALT"]]<-"Aminotransferase, alanine (ALT)"
-    potential_settings[["measure_values"]][["AST"]]<-"Aminotransferase, aspartate (AST)"
-    potential_settings[["measure_values"]][["TB"]]<-"Total Bilirubin"
-    potential_settings[["measure_values"]][["ALP"]]<-"Alkaline phosphatase (ALP)"
+  if (standard == "None") {
+    standard_low <- "default"
   }
   
-  
-  if(partial) {
+  # Build empty settings list
+  potential_settings <- list()
+  for ( i in 1:length(hierarchical_metadata) ) {
     
-    settings_names <- names(settings)
-    
-    potential_names <- names(potential_settings)
-    
-    for(i in 1:length(settings)) {
-      if (potential_settings[i] %in% partial_cols) {
-            settings[[which(settings_names == potential_names[i])]] <- potential_settings[[i]]
-          }
+    # Handle settings with one level
+    if (length(hierarchical_metadata[[i]]) == 1) {
+
+      #Handle Nulls
+       if (is.null(parse_expr(metadata[[standard_low]][i]))) {
+         potential_settings[metadata$text_key[i]] = list(parse_expr(metadata[[standard_low]][i]))
+         
+        # Handle C()  - not implemented for level two yet
+       } else if (typeof(parse_expr(metadata[[standard_low]][i])) == "language") {
+         potential_settings[[metadata$text_key[i]]] = as.character(parse_expr(metadata[[standard_low]][i]))[-1]  #Need the -1 to remove the unwelcome "c" that comes with this method
+         
+       } else {
+        potential_settings[metadata$text_key[[i]]] = parse_expr( metadata[[standard_low]][i])
+      }
+            
+      # Handle settings with two levels
+    } else if (length(hierarchical_metadata[[i]]) == 2){
+      
+      #Create list if it does not exist
+      if (!is.list(potential_settings[[hierarchical_metadata[[i]][1]]])) { potential_settings[[hierarchical_metadata[[i]][1]]] = list() } #Need to make list if it doesnt exist since its two-level
+      
+      #Handle Nulls
+       if (is.null(parse_expr(metadata[[standard_low]][i]))) {
+         potential_settings[[hierarchical_metadata[[i]][1]]][hierarchical_metadata[[i]][2]] = list(parse_expr(metadata[[standard_low]][i]))
+         
+       } else {
+        potential_settings[[hierarchical_metadata[[i]][1]]][[hierarchical_metadata[[i]][2]]] = parse_expr( metadata[[standard_low]][i])
+      }
+            
+    } else{
+      stop("Three level setting nests are not currently supported")
```

Comment:
This is probably a problem ... `settings.filters[0].label` is a valid setting for example